src/latexify/codegen/algorithmic_codegen.py

```diff
@@ -0,0 +1,109 @@
+"""Codegen for single algorithms."""
+
+from __future__ import annotations
+
+import ast
+
+from latexify import exceptions
+from latexify.codegen import codegen_utils, expression_codegen, identifier_converter
+
+
+class AlgorithmicCodegen(ast.NodeVisitor):
+    """Codegen for single algorithms.
+
+    This codegen works for Module with single FunctionDef node to generate a single
+    LaTeX expression of the given algorithm.
+    """
+
+    _identifier_converter: identifier_converter.IdentifierConverter
+
+    def __init__(
+        self, *, use_math_symbols: bool = False, use_set_symbols: bool = False
+    ) -> None:
+        """Initializer.
+
+        Args:
+            use_math_symbols: Whether to convert identifiers with a math symbol surface
+                (e.g., "alpha") to the LaTeX symbol (e.g., "\\alpha").
+            use_set_symbols: Whether to use set symbols or not.
+        """
+        self._expression_codegen = expression_codegen.ExpressionCodegen(
+            use_math_symbols=use_math_symbols, use_set_symbols=use_set_symbols
+        )
+        self._identifier_converter = identifier_converter.IdentifierConverter(
+            use_math_symbols=use_math_symbols
+        )
+
+    def generic_visit(self, node: ast.AST) -> str:
+        raise exceptions.LatexifyNotSupportedError(
+            f"Unsupported AST: {type(node).__name__}"
+        )
+
+    def visit_Assign(self, node: ast.Assign) -> str:
+        """Visit an Assign node."""
+        operands: list[str] = [
+            self._expression_codegen.visit(target) for target in node.targets
+        ]
+        operands.append(self._expression_codegen.visit(node.value))
+        operands_latex = r" \gets ".join(operands)
+        return rf"\State ${operands_latex}$"
+
+    def visit_Expr(self, node: ast.Expr) -> str:
```

Comment:
Required for something like `if True: x`. Have also added `visit_Expr` to `expression_codegen.py`.

Reply:
`Expr` is `stmt` that owns only one expression. I think it should be procecced by `stmt` codegen.

Reply:
ah yes you're right, fixing that